app/models/SubmissionType.scala

```diff
@@ -22,32 +22,41 @@ import uk.gov.hmrc.govukfrontend.views.Aliases.Text
 import uk.gov.hmrc.govukfrontend.views.viewmodels.radios.RadioItem
 import views.ViewUtils.hint
 
-sealed trait WhatDoYouWantToDo
+sealed trait SubmissionType
 
-object WhatDoYouWantToDo extends Enumerable.Implicits[WhatDoYouWantToDo] {
+object SubmissionType extends Enumerable.Implicits[SubmissionType] {
 
-  case object CreateOption extends WithName("createOption") with WhatDoYouWantToDo
+  case object CreateCase extends WithName("createCase") with SubmissionType
 
-  case object UpdateOption extends WithName("updateOption") with WhatDoYouWantToDo
+  case object UpdateCase extends WithName("updateCase") with SubmissionType
 
-  case object CancelOption extends WithName("cancelOption") with WhatDoYouWantToDo
+  case object CancelCase extends WithName("cancelCase") with SubmissionType
 
-  val values: Seq[WhatDoYouWantToDo] = Seq(
-    CreateOption, UpdateOption, CancelOption
+  val values: Seq[SubmissionType] = Seq(
+    CreateCase, UpdateCase, CancelCase
   )
 
-  def options(form: Form[_])(implicit messages: Messages): Seq[RadioItem] = values.map {
+
+  def options(form: Form[_], cancelCase: Boolean, updateCase: Boolean)(implicit messages: Messages): Seq[RadioItem] = {
+    val values = (cancelCase, updateCase) match {
+      case (true, true) => Seq(CreateCase, UpdateCase, CancelCase)
+      case (true, false) => Seq(CreateCase, CancelCase)
+      case (false, true) => Seq(CreateCase, UpdateCase)
+      case _ => Seq.empty
+    }
```

Comment:
I think the last case should be `_ => Seq(CreateCase)` because we always want to show at least one button
Also, I think this could be written more tersely if you want, the existing code is OK though
```scala
Seq(CreateCase) ++ Seq(UpdateCase).filter(_ => updateCase) ++ Seq(CancelCase).filter(_ => cancelCase)
```